code/modules/admin/greyscale_modify_menu.dm

```diff
@@ -69,25 +95,42 @@
 			if(new_color)
 				split_colors[group] = new_color
 				refresh_preview()
+
 		if("apply")
+			target.set_greyscale_config(config.type, update=FALSE)
 			target.greyscale_colors = "" // We do this to force an update, in some cases it will think nothing changed when it should be refreshing
 			target.set_greyscale_colors(split_colors)
+
 		if("refresh_file")
 			SSgreyscale.RefreshConfigsFromFile()
 			refresh_preview()
 
+		if("change_dir")
+			sprite_dir = text2dir(params["new_sprite_dir"])
+			refresh_preview()
+
 /datum/greyscale_modify_menu/proc/ReadColorsFromString(colorString)
 	var/list/raw_colors = splittext(colorString, "#")
 	split_colors = list()
 	for(var/i in 2 to length(raw_colors))
 		split_colors += "#[raw_colors[i]]"
 
 /datum/greyscale_modify_menu/proc/refresh_preview()
-	var/list/data = SSgreyscale.configurations["[target.greyscale_config]"].GenerateDebug(split_colors.Join())
+	for(var/i in length(split_colors) + 1 to config.expected_colors)
+		split_colors += rgb(100, 100, 100)
+	var/list/used_colors = split_colors.Copy(1, config.expected_colors+1)
```

Comment:
why the + 1 here?

Reply:
![image](https://user-images.githubusercontent.com/1234602/116770334-0df10f80-aa11-11eb-99eb-e21fafecf0f4.png)

We want an amount of colors equal to the expected colors, this chunk of code here both makes sure it's not too short and cuts off extra colors if too long. `Copy(1, x)` would return a list with x-1 items.

Reply:
ah, excuse me